disco/extensions/upgrade_simulation/upgrades/voltage_upgrade_functions.py

```diff
@@ -907,7 +907,11 @@ def determine_new_regulator_upgrades(voltage_config, buses_with_violations, volt
         best_setting_so_far = "after_addition_new_regcontrol"
     else: 
         new_reg_upgrade_commands = []
-        dss_commands_list = list(set(dss_commands_list) - set(regcontrol_settings_commands_list) - set(regcontrol_cluster_commands))
+        remove_commands_list = regcontrol_settings_commands_list + regcontrol_cluster_commands
+        if remove_commands_list:
+            for x in dss_commands_list:
+                if x in remove_commands_list:
+                    dss_commands_list.remove(x)

```

Comment:
This could be problematic because you are mutating a list while iterating over it. If you don't care about the order of items, I would use a set, as in `set(dss_commands_list).difference(remove_commands_list)`

Reply:
It was using set before, but the order is important - else it will raise an error. Is there a better way to remove while preserving order?

Reply:
You would need to build a temporary list of the names or indexes to remove and then iterate over that to perform the removals.